lib/ansible/modules/windows/win_partition.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Varun Chopra (@chopraaa)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_partition
+version_added: '2.8'
+short_description: Creates, changes and removes partitions on Windows Server
+description:
+- The M(win_partition) module can create, modify or delete a partition on a disk
+options:
+  state:
+    description:
+      - Used to specify the state of the partition. Use C(absent) to specify if a partition should be removed
+        and C(present) to specify if the partition should be created or updated.
+    type: str
+    choices: [ absent, present]
+    default: present
+  drive_letter:
+    description:
+      - Used for accessing partitions if I(disk_number) and I(partition_number) are not provided.
+      - Use C(auto) for automatically assigning a drive letter, or a letter A-Z for manually assigning a drive letter to a new partition.
+        If not specified, no drive letter is assigned when creating a new partition.
+    type: str
+  disk_number:
+    description:
+      - Disk number is mandatory for creating new partitions.
+      - A combination of I(disk_number) and I(partition_number) can be used to specify the partition instead of I(drive_letter) if required.
+    type: int
+  partition_number:
+    description:
+      - Used in conjunction with I(disk_number) to uniquely identify a partition.
+    type: int
+  partition_size:
+    description:
+      - Specify size of the partition in GB. Use -1 to specify maximum supported size.
```

Comment:
I initially started out with using `B` since it allowed more flexibility, but [Ansible assigns integers as `int32` by default](https://github.com/ansible/ansible/blob/f13091d142d3eea2ce94bf0c6b164039472c32a5/lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1#L234). I suppose option #2 could allow us to be more flexible in this case.

Reply:
You would have to remove the type if you don’t want int32. We deal with unsigned ints that are 64 bits in length in win_regedit and have to rely on the manual parsing.

Reply:
Great, that makes sense. Will wait for Dag to weigh in before I make any changes. 

Also,

>  allow people to add the unit as a prefix

I may be misinterpreting but I think you're implying `GiB 20` and not `20 GiB`?

Reply:
Ah yes sorry, ~~prefix~~ suffix

Reply:
@jborean93 My preference here would be to allow to put the unit as part of the value (which obviously means we ought to have helper-functions to translate the values to the needed format).

If the unit is missing, I would assume it is in bytes. But I can see how that could be difficult in cases where values in bytes make little sense. So maybe the module could provide a different default unit (and specify it in the parameter docs).

I am not a big fan for having the unit in the parameter name, or having a separate parameter that stores the unit (usually done to avoid the need for parsing). The unit as part of the value is definitely the most user-friendly and concise option.